lib/ansible/plugins/inventory/yaml.py

```diff
@@ -18,10 +18,19 @@
         - It takes the place of the previously hardcoded YAML inventory.
         - To function it requires being whitelisted in configuration.
     options:
-        yaml_extensions:
-            description: list of 'valid' extensions for files containing YAML
-            type: list
-            default: ['.yaml', '.yml', '.json']
+      yaml_extensions:
+        description: list of 'valid' extensions for files containing YAML
+        type: list
+        default: ['.yaml', '.yml', '.json']
+        env:
+          - name: ANSIBLE_YAML_FILENAME_EXT
+          - name: ANSIBLE_INVENTORY_PLUGIN_EXTS
+        ini:
+          - key: yaml_valid_extensions
```

Comment:
perhaps order these the same: key then section or section then key.

Reply:
the list order is significant, the dictionary order is not